extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -378,18 +362,45 @@ export class AmpIframe extends AMP.BaseElement {
   /**
    * Updates the element's dimensions to accommodate the iframe's
    *    requested dimensions.
-   * @param {number|undefined} newWidth
-   * @param {number|undefined} newHeight
+   * @param {number|undefined} height
+   * @param {number|undefined} width
    * @private
    */
-  updateSize_(newHeight, newWidth) {
+  updateSize_(height, width) {
     if (!this.isResizable_) {
       user.warn(TAG_,
           'ignoring embed-size request because this iframe is not resizable',
           this.element);
       return;
     }
-    this.attemptChangeSize(newHeight, newWidth);
+
+    let newHeight, newWidth;
+    if (height !== undefined) {
+      newHeight = Math.max(
+        (this.element./*OK*/offsetHeight - this.iframe_./*OK*/offsetHeight)
+            + height,
+        height);
+    }
+    if (width !== undefined) {
+      newWidth = Math.max(
+        (this.element./*OK*/offsetWidth - this.iframe_./*OK*/offsetWidth)
+            + width,
+        width);
+    }
+
+    if (newHeight !== undefined || newWidth !== undefined) {
+      this.attemptChangeSize(newHeight, newWidth, () => {
+        this.iframe_.height = height;
```

Comment:
If it works without in manual testing, lets kill it.

On Thu, Jun 30, 2016 at 5:49 PM, Sriram Krishnan <notifications@github.com>
wrote:

> In extensions/amp-iframe/0.1/amp-iframe.js
> <https://github.com/ampproject/amphtml/pull/3858#discussion_r69233702>:
>
> > +    if (height !== undefined) {
> > +      newHeight = Math.max(
> > +        (this.element./*OK*/offsetHeight - this.iframe_./*OK*/offsetHeight)
> > +            + height,
> > +        height);
> > +    }
> > +    if (width !== undefined) {
> > +      newWidth = Math.max(
> > +        (this.element./*OK*/offsetWidth - this.iframe_./*OK*/offsetWidth)
> > +            + width,
> > +        width);
> > +    }
> > +
> > +    if (newHeight !== undefined || newWidth !== undefined) {
> > +      this.attemptChangeSize(newHeight, newWidth, () => {
> > +        this.iframe_.height = height;
>
> lets file a bug and have @dvoytenko <https://github.com/dvoytenko> weigh
> in , add a TODO for yourself with the bug no and lets get this in.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/3858/files/fb7c862d5939e69c2b730fdb2e7de5fcdfc6d64a#r69233702>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAFeT2dDwTF0RP3TFdocCGL-yZcDu3o0ks5qRGQogaJpZM4JCsEC>
> .
>